server/cast/utils.go

```diff
@@ -19,9 +19,19 @@ import (
 	"strings"
 	"unicode/utf8"
 
+	"gopkg.in/src-d/go-errors.v1"
+
 	pgtypes "github.com/dolthub/doltgresql/server/types"
 )
 
+var (
+	// errCannotCast is returned when a cast from one type to another is not possible.
+	errCannotCast = errors.NewKind("cannot cast type %s to %s")
+
+	// errOutOfRange is returned when a value is out of range for a given type.
+	errOutOfRange = errors.NewKind("%s out of range")
```

Comment:
What do you think of something like this for the cast errors? Noticed a lot were repeated. Can update the others if you like

Reply:
I hadn't used these since we don't use the proper error model just yet (lack of error ID), and was planning on adding them in the relatively near future before it gets out of hand. It's fine either way though, so its your preference.